hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCancelAirdropHandler.java

```diff
@@ -50,14 +51,13 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {}
 
     @Override
-    public void handle(@NonNull final HandleContext context) throws HandleException {
-        var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
-        validateTrue(tokensConfig.cancelTokenAirdropEnabled(), NOT_SUPPORTED);
-    }
+    public void handle(@NonNull final HandleContext context) throws HandleException {}
 
     @NonNull
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
-        return Fees.FREE;
+        var tokensConfig = feeContext.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.cancelTokenAirdropEnabled(), NOT_SUPPORTED);
+        return feeContext.feeCalculatorFactory().feeCalculator(SubType.DEFAULT).calculate();
```

Comment:
We need to also add `VPT` section for this, just like we did in `TokenMintHandler`. Because number of signatures should increase the price

Reply:
Added 👍